tc-tests-utils.sh

```diff
@@ -712,3 +732,63 @@ package_native_client_ndk()
     -C ${deepspeech_dir}/native_client/kenlm/ README.mozilla \
     | pixz -9 > "${artifacts_dir}/${artifact_name}"
 }
+
+android_setup_emulator()
+{
+  if [ -z "${ANDROID_SDK_HOME}" ]; then
+    echo "No Android SDK home available, aborting."
+    exit 1
+  fi;
+
+  if [ -z "$1" ]; then
+    echo "No ARM flavor, please give one."
+    exit 1
+  fi;
+
+  flavor=$1
+
+  export PATH=${ANDROID_SDK_HOME}/tools/bin/:${ANDROID_SDK_HOME}/platform-tools/:$PATH
+  export DS_BINARY_PREFIX="adb shell LD_LIBRARY_PATH=${ANDROID_TMP_DIR}/ds/ ${ANDROID_TMP_DIR}/ds/"
+
+  sdkmanager --verbose --update
+  sdkmanager --verbose --install "platform-tools"
+  sdkmanager --verbose --install "emulator"
+  sdkmanager --verbose --install "platforms;android-25"
+  sdkmanager --verbose --install "system-images;android-25;google_apis;${flavor}"
+  avdmanager create avd --name "ds-pixel" --device 17 --package "system-images;android-25;google_apis;${flavor}"
+
+  pushd ${ANDROID_SDK_HOME}
+    ./tools/emulator -verbose -avd ds-pixel -no-skin -no-audio -no-window &
+    emulator_rc=$?
+    export ANDROID_DEVICE_EMULATOR=$!
+  popd
+
+  if [ "${emulator_rc}" -ne 0 ]; then
+    echo "Error starting Android emulator, aborting."
+    exit 1
+  fi;
+
+  adb wait-for-device
+
+  adb shell id
+  adb shell cat /proc/cpuinfo
+
+  adb shell mkdir ${ANDROID_TMP_DIR}/ds/
+  adb push ${TASKCLUSTER_TMP_DIR}/ds/* ${ANDROID_TMP_DIR}/ds/
+
+  adb push \
+    ${TASKCLUSTER_TMP_DIR}/${model_name} \
+    ${TASKCLUSTER_TMP_DIR}/LDC93S1.wav \
+    ${TASKCLUSTER_TMP_DIR}/alphabet.txt \
+    ${ANDROID_TMP_DIR}/ds/
+}
+
+android_stop_emulator()
+{
+  if [ -z "${ANDROID_DEVICE_EMULATOR}" ]; then
+    echo "No ANDROID_DEVICE_EMULATOR"
+    exit 1
+  fi;
+
+  kill -9 ${ANDROID_DEVICE_EMULATOR}
```

Comment:
Can SIGKILL cause problems when shutting down the emulators? Just wondering if this is a potential source of intermittents.

Reply:
it should not be an issue